yarn-project/simulator/src/public/avm_executor.test.ts

```diff
@@ -35,34 +36,16 @@ describe('AVM WitGen and Proof Generation', () => {
     header = makeHeader(randomInt(1000000));
   }, 10000);
 
-  it('Should prove valid execution of bytecode that performs addition', async () => {
-    const args: Fr[] = [new Fr(1), new Fr(2)];
-    // Bytecode for the following contract is encoded:
-    // const bytecode = encodeToBytecode([
-    //    new CalldataCopy(/*indirect=*/ 0, /*cdOffset=*/ 0, /*copySize=*/ 2, /*dstOffset=*/ 0),
-    //    new Add(/*indirect=*/ 0, TypeTag.FIELD, /*aOffset=*/ 0, /*bOffset=*/ 1, /*dstOffset=*/ 2),
-    //    new Return(/*indirect=*/ 0, /*returnOffset=*/ 2, /*copySize=*/ 1),
-    // ]);
-    const bytecode: Buffer = Buffer.from('IAAAAAAAAAAAAgAAAAAAAAYAAAAAAAAAAQAAAAI5AAAAAAIAAAAB', 'base64');
-    publicContracts.getBytecode.mockResolvedValue(bytecode);
-    const executor = new PublicExecutor(publicState, publicContracts, commitmentsDb, header);
-    const functionData = FunctionData.empty();
-    const execution: PublicExecution = { contractAddress, functionData, args, callContext };
-    const [proof, vk] = await executor.getAvmProof(execution);
-    const valid = await executor.verifyAvmProof(vk, proof);
-    expect(valid).toBe(true);
-  });
-
-  // This is skipped as we require MOV to be implemented in the AVM
-  it.skip('Should prove valid execution contract function that performs addition', async () => {
-    const args: Fr[] = [new Fr(1), new Fr(2)];
-
+  it('Should prove valid execution contract function that performs addition', async () => {
     const addArtifact = AvmTestContractArtifact.functions.find(f => f.name === 'add_args_return')!;
     const bytecode = addArtifact.bytecode;
     publicContracts.getBytecode.mockResolvedValue(bytecode);
+
     const functionData = FunctionData.fromAbi(addArtifact);
-    const execution: PublicExecution = { contractAddress, functionData, args, callContext };
+    const args: Fr[] = [new Fr(99), new Fr(12)];
+    const context = initContext({ env: initExecutionEnvironment({ calldata: args }) });
```

Comment:
Maybe add a comment on why we are doing this dance to get the calldata back, and that this would make more sense if you were simulating?